cmd/zap/cmd/vector.go

```diff
@@ -128,6 +128,9 @@ func decodeSection(data []byte, start uint64) (int, int, map[int64][]uint32, *fa
 	_, n = binary.Uvarint(data[pos : pos+binary.MaxVarintLen64])
 	pos += n
 
+	_, n = binary.Uvarint(data[pos : pos+binary.MaxVarintLen64])
+	pos += n
+
```

Comment:
I realize this code duplication wasn't introduced here, but this PR is sure building on it:
Lines 125 through 133 show so much repetition. For starters -
```
var n int
for i := 0 ; i < 3; i++ {
    _, n = binary.Uvarint(data[pos : pos+binary.MaxVarintLen64])
    pos += n
}
```

Also the comment above needs fixing and additional commentary around why this needs to occur 3 times.